conans/client/tools/net.py

```diff
@@ -67,6 +61,19 @@ def download(url, filename, verify=True, out=None, retry=None, retry_wait=None,
     retry_wait = retry_wait if retry_wait is not None else 5
 
     downloader = FileDownloader(requester=requester, output=out, verify=verify, config=config)
-    downloader.download(url, filename, retry=retry, retry_wait=retry_wait, overwrite=overwrite,
-                        auth=auth, headers=headers)
+    checksum = sha256 or sha1 or md5
+    if config and config.download_cache and checksum:
```

Comment:
This deserves a comment (probably in the docs): the download-cache will only be used if a checksum is provided.